go/trace/trace.go

```diff
@@ -97,17 +107,43 @@ func RegisterSpanFactory(sf SpanFactory) {
 	spanFactory = sf
 }
 
-type fakeSpanFactory struct{}
-
-func (fakeSpanFactory) New(parent Span) Span                                         { return fakeSpan{} }
-func (fakeSpanFactory) FromContext(ctx context.Context) (Span, bool)                 { return nil, false }
-func (fakeSpanFactory) NewContext(parent context.Context, span Span) context.Context { return parent }
+var spanFactory SpanFactory = fakeSpanFactory{}
 
-// fakeSpan implements Span with no-op methods.
-type fakeSpan struct{}
+var (
+  tracingServer = flag.String("tracer", "noop", "tracing service to use.")
+)
 
-func (fakeSpan) StartLocal(string)            {}
-func (fakeSpan) StartClient(string)           {}
-func (fakeSpan) StartServer(string)           {}
-func (fakeSpan) Finish()                      {}
-func (fakeSpan) Annotate(string, interface{}) {}
+// StartTracing enables tracing for a named service
+func StartTracing(serviceName string) io.Closer {
```

Comment:
I think I need help with the commenting of this method. 

This is not about starting a span, it's about starting up the tracing system. It's done like this so that executables can tell the tracing system which name they are under ("vtgate", "vtclient", "vtwork", etc), and a chance of explicitly closing the tracing system so that everything is flushed nicely.